### PR TITLE
Test the detection of known Go binaries

### PR DESCRIPTION
**Pull request by Arker123**
This PR includes tests to verify the detection of known strings in various versions of Go binaries, which are extracted from zip archives.

Solves #807

### CODE REVIEW

**Comment by Arker123** [comment]
Once #842 is merged, let's proceed with merging this.

File: tests/test_language_go_known_binary.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+                / "go-binaries-all-versions"
+                / "bin"
+                / "extracted_64"
+                / zip_info.filename
+            ).resolve()
+            print(extracted_file_path)
+            extracted_file = zip_ref.extract(zip_info, path=extracted_file_path.parent)
+            extracted_files.append(extracted_file)
+
+    print(extracted_files)
```
Comment (williballenthin) [inline tests/test_language_go_known_binary.py:45]:
```suggestion
```

File: tests/test_language_go_known_binary.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+                CD
+                / "data"
+                / "language"
+                / "go"
+                / "go-binaries-all-versions"
+                / "bin"
+                / "extracted_32"
+                / zip_info.filename
+            ).resolve()
+            print(extracted_file_path)
```
Comment (williballenthin) [inline tests/test_language_go_known_binary.py:86]:
```suggestion
```

File: tests/test_language_go_known_binary.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+            extracted_file = zip_ref.extract(zip_info, path=extracted_file_path.parent)
+            extracted_files.append(extracted_file)
+
+    print(extracted_files)
```
Comment (williballenthin) [inline tests/test_language_go_known_binary.py:90]:
```suggestion
```

**Review by williballenthin — CHANGES_REQUESTED** [review]
remove debugging `print` statements and then ready to go

File: tests/test_language_go_known_binary.py [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+    yield
+
+    # Clean up - remove the extracted files after the test finishes
+    for extracted_file in extracted_files:
+        pathlib.Path(extracted_file).unlink()
+    pathlib.Path(extracted_file_path.parent).rmdir()
+
+
+@pytest.fixture(scope="module")
+def extract_files_32():
```
Comment (mr-tz) [inline tests/test_language_go_known_binary.py:53]: can you unify this and `extract_files_64` be into one function?

File: tests/test_language_go_known_binary.py [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+        ("data/language/go/go-binaries-all-versions/bin/extracted_32/main1.14.exe"),
+        ("data/language/go/go-binaries-all-versions/bin/extracted_32/main1.15.exe"),
+        ("data/language/go/go-binaries-all-versions/bin/extracted_32/main1.16.exe"),
+        ("data/language/go/go-binaries-all-versions/bin/extracted_32/main1.17.exe"),
+        ("data/language/go/go-binaries-all-versions/bin/extracted_32/main1.18.exe"),
+        ("data/language/go/go-binaries-all-versions/bin/extracted_32/main1.19.exe"),
+        ("data/language/go/go-binaries-all-versions/bin/extracted_32/main1.20.exe"),
+    ],
+)
+def test_language_detection_32(binary_file, extract_files_32):
```
Comment (mr-tz) [inline tests/test_language_go_known_binary.py:146]: these test names don't appear to be indicating what's actually tested here, please update them

**Review by mr-tz — CHANGES_REQUESTED** [review]

**Review by mr-tz — APPROVED** [review]
great, thanks!

**Pull request merged**